### ISSUE TITLE
Group directive not working when user belongs to many groups

### ISSUE THREAD

**Issue by lapause** [mysecureshell/mysecureshell#100]
Version used: 2.0

I have users that belong to a large number of unix groups (250+).
Directives in `<Group>` sections of the configuration that should be applied are ignored.
If I create a `<User>` section specific to one of those users and put the directives there, everything works as expected.

E.g: user `test`, member of **276 groups** including **shell**

```
<Default>
   (...)
   Shell /bin/false
</Default>

<Group shell>
   # User test is member of shell, should have shell access but not working
   Shell /bin/bash
</Group>

# If a use a User specific section, it's working as expected
<User test>
   Shell /bin/bash
</User>
```

**Comment by Teka101** [comment]
Hello,

When i check the code of MSS, there is a current limitation of 63 groups for an user :(

**Comment by lapause** [comment]
I think I might have found a related issue.

Accessing via SFTP a directory _projects_ with hundreds of subdirectories owned by different users/groups (groups to which the SFTP user belongs), directory _projects_ appears empty.
A direct SFTP `cd` command to access a subdirectory returns an error.

FYI on a legacy server running v1.25rc5 of MySecureShell, the same kind of tree structure can be browsed without issue.

As I have never experienced this odd behaviour and everything works fine in directories with less different owners/groups, I think it might be related to the main issue. Sadly I can't think of a way to bypass this one. Tell me if you want me to open another issue.

**Comment by Teka101** [comment]
Yes, exactly, two issues for this limitation... no need to open another issue :)

**Comment by iv9meesaiL5kahv0** [comment]
May I ask why this bug is still open after more than 4 years? Is this just about changing a number on line 39 in file mysecureshell/SftpServer/Access.c or is there a bigger picture that I am missing?

In general, why is there a limit at all? Or would it be an option to make this threshold value configurable via the config file?
